src/app/modules/item/components/sub-skills/sub-skills.component.ts

```diff
@@ -24,18 +25,43 @@ interface SubSkillAdditions {
 export class SubSkillsComponent implements OnChanges, OnDestroy {
   @Input() itemData?: ItemData;
 
-  state: 'loading' | 'error' | 'ready' = 'loading';
-  children: (ItemChild&SubSkillAdditions)[] = [];
-
-  private subscription?: Subscription;
+  private readonly params$ = new ReplaySubject<{ id: string, attemptId: string }>(1);
+  readonly state$ = this.params$.pipe(
+    distinctUntilChanged((a, b) => a.id === b.id && a.attemptId === b.attemptId),
+    switchMap(({ id, attemptId }) =>
+      this.getItemChildrenService.get(id, attemptId).pipe(
+        map(children =>
```

Comment:
same old same old :smile: 
I point them out to make sure we don't forget one, that's all :grimacing:

Reply:
Fixed